src/aggregate.h

```diff
@@ -187,6 +187,7 @@ class StructDeclaration : public AggregateDeclaration
     // (e.g. TypeidExp, NewExp, ArrayLiteralExp, etc) request its TypeInfo.
     // For those, today TypeInfo_Struct is generated in COMDAT.
     bool requestTypeInfo;
+    bool cppmangleAsClass;      // true if this struct should be mangled as class (VS only)
```

Comment:
Also, we can merge the two opposite meaning flags into an `AggregateDeclaration.cppMangleAs`.